src/backend/catalog/index.c

```diff
@@ -2927,9 +2930,16 @@ IndexBuildAppendOnlyRowScan(Relation parentRelation,
 	predicate = (List *)
 		ExecPrepareExpr((Expr *)indexInfo->ii_Predicate, estate);
 	
+	/*
+	 * Regular MVCC snapshot, which may be taken in IndexBuildScan function,
+	 * can hide newly globally inserted tuples from global index build process,
+	 * because it's outdated in such case (see dispatching part of DefineIndex
+	 * for more). Since this, we need to use SnapshotSelf to get the actual
+	 * visibility map.
+	 */
```

Comment:
IMO when we mention here `MVCC snapshot` and `SnapshotSelf` we have to add that they regard to snapshot for metadata.

> ... to get the actual visibility map

"... to get actual metadata (pg_aoseg, in particular)". `visibility map` is not correct to use here